http/vertx-web-validation/src/main/java/io/quarkus/ts/vertx/web/validation/ValidationHandlerOnRoutes.java

```diff
@@ -0,0 +1,159 @@
+
+                        String errorMessage = routingContext.failure().toString();
+                        routingContext.response()
+                                .setStatusCode(HttpResponseStatus.BAD_REQUEST.code())
+                                .putHeader(HttpHeaders.CONTENT_TYPE, MediaType.APPLICATION_JSON)
+                                .end(String.format(ERROR_MESSAGE, errorMessage));
+                    }
+                });
+        // Create a ValidationHandlerBuilder with explodedParam and arraySchema to filter by array items
+        ObjectSchemaBuilder bodySchemaBuilder = objectSchema()
+                .property("shoppingListName", stringSchema());
+        ValidationHandlerBuilder
+                .create(schemaParser)
+                .body(Bodies.json(bodySchemaBuilder));
+        router.get("/filterByArrayItem")
+                .handler(
+                        ValidationHandlerBuilder
+                                .create(schemaParser)
+                                .queryParameter(Parameters.explodedParam("shoppingArray", arraySchema().items(stringSchema())))
+                                .body(Bodies.json(bodySchemaBuilder))
+                                .build())
+                .handler(routingContext -> {
+                    RequestParameters parameters = routingContext.get(ValidationHandler.REQUEST_CONTEXT_KEY);
+                    JsonArray myArray = parameters.queryParameter("shoppingArray").getJsonArray();
+                    // Retrieve the list of all shoppingLists
+                    List<ShoppingList> shoppingLists = fetchProductDetailsFromArrayQuery(myArray);
+
+                    routingContext.response().putHeader(HttpHeaders.CONTENT_TYPE, MediaType.APPLICATION_JSON)
+                            .end(Json.encodeToBuffer(shoppingLists));
+                });
+        // Let's allow to create a new item
+        router.post("/createShoppingList").handler(
+                ValidationHandlerBuilder
+                        .create(schemaParser)
+                        .predicate(RequestPredicate.BODY_REQUIRED)
+                        .queryParameter(param("shoppingListName", stringSchema()))
+                        .queryParameter(param("shoppingListPrice", numberSchema().with(maximum(100))))
+                        .build())
+                .handler(routingContext -> {
+                    routingContext.response().setStatusCode(HttpResponseStatus.OK.code()).end("Shopping list created");
+                });
+
+    }
+
+    public List<ShoppingList> fetchProductDetailsFromArrayQuery(JsonArray myArray) {
+        return shopResource.get().stream()
+                .filter(shoppingList -> myArray.contains(shoppingList.getName()))
+                .collect(Collectors.toList());
+    }
+
+    public String fetchProductDetailsFromQuery(String name, Double price) {
+        return shopResource.get().stream()
```

Comment:
sorry I did not mention this until now, so you basically are using resource as a bean and it is never called, hmm, in which case I do not see a point of having it Path annotation etc.

Is there a case when response is not sent by your route handler but it actually goes to RESTEasy Reactive?

Reply:
I use ShopResource as a bean and the `@Path("/shoppinglist") `is tested through the test checkShoppingListUrl().
is it what you mean?